pytest-slc/exasol/pytest_slc/__init__.py

```diff
@@ -65,16 +64,17 @@ def export_slc(slc_builder, export_slc_async) -> Path | None:
 
 @pytest.fixture(scope="session")
 def upload_slc(slc_builder, export_slc,
-               backend_aware_database_params, backend_aware_bucketfs_params):
+               pyexasol_connection, backend_aware_bucketfs_params):
     """
-    The fixture uploads language container to a database, according to the selected
-    backends.
+    The fixture provides a function uploading the language container to a database,
+    according to the selected backends.
     """
-    if (slc_builder is not None) and (export_slc is not None):
-        pyexasol_connection = pyexasol.connect(**backend_aware_database_params)
-        bucketfs_path = bfs.path.build_path(**backend_aware_bucketfs_params,
-                                            path=BFS_CONTAINER_DIRECTORY)
-        deployer = LanguageContainerDeployer(pyexasol_connection=pyexasol_connection,
-                                             bucketfs_path=bucketfs_path,
-                                             language_alias=slc_builder.language_alias)
-        deployer.run(container_file=export_slc, alter_system=True, allow_override=True)
+    def func(language_alias: str) -> None:
```

Comment:
I guess, you changed it to that, such that we can isolate the aliases per test, if it necessary. I could assume, in case of the deployer tests and the extension tests that could be necessary. Or, did you think of a different reason.

Reply:
My question was related to the function and not the Pyexasol connection

Reply:
But good point , maybe we need a fixture that returns a function that activates the SLC for a connection

Reply:
> We used to take the language_alias from the Builder ...  language_alias is not actually required to build a language container ... But for the deployer. ... 

Ah - very good, thank you!

Reply:
Decision:
- Rename fixture upload_slc to deploy_slc
- Add fixture deployed_slc which uses a dummy fixture for alias which needs be overwritten by the extension
- the function returned by deploy_slc fixture uses a state variable stored in deploy_slc fixture to upload the slc only once and only activate it in subsequent calls